Add EIP: EIP tagging (#11545)
ethereum/EIPs: EIPS/eip-8235.md

Reviewer: So you'll want to word this whole thing a bit differently, if the intent is to create an EIP-5757 or EIP-5069 -style appendix for EIP-1.

Instead of writing this like a proposal introducing a new rule, write as if the rule already applies.

Author: I went through and changed EIP-1 (and the EIP-template)

> ---

Reviewer 2: ```suggestion
---
eip: 8235
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.

Please also update the filename.

Reviewer 3: > ```suggestion
> 111
> eip: 8235
> ```
> Assigning next sequential EIP/ERC/RIP number.
> Numbers are assigned by editors & associates.
> 
> Please also update the filename.

---
eip: 8235
title: EIP tagging
description: Adds required tags for EIPs
author: James Kempton (@SirSpudlington), et al.
discussions-to: https://ethereum-magicians.org/t/meta-eip-8235-eip-tagging/28275
status: Draft
type: Meta
created: 2026-04-18
requires: 1
---

## Abstract

> Introduce a was to tag EIPs with well-known specifics on what the EIP affects and defines governance rules on the addition and removal of these tags.

Reviewer: ```suggestion
Introduce a way to tag EIPs with well-known specifics on what the EIP affects and defines governance rules on the addition and removal of these tags.
```

> Introduce a was to tag EIPs with well-known specifics on what the EIP affects and defines governance rules on the addition and removal of these tags.

Reviewer: So instead of what you have here, you'd write something like:

```suggestion
EIPs and ERCs are categorized according to a set of well-known tags describing what the proposal affects. This document defines the tags and the governance rules for curating the list.
```

## Specification

### Tag structure

EIP tags must follow the structure of `namespace:category[/subcategory]`. The `[/subcategory]` section of a tag is optional, and may be omitted if an EIP broadly covers a topic.
Tags should be appended on a per-tag basis, no two tags are mutually exclusive.

Core EIPs must include the minimal number of tags that can fully cover the proposal. Tags are optional for non-core EIPs but if included, must be comprehensive and minimal.

> EIPs must define tags within the EIP preamble under the `tags` key.

Reviewer: These paragraphs don't belong in the "tag structure" section, because they don't really describe the structure of tags.

Honestly, I'd put both of these in EIP-1 and link from there to this document.
